File: packages/uns-core/src/uns/uns-path.ts
const normalizeSegment = (value: string): string => value.trim().replace(/^\/+|\/+$/g, "");

export function buildUnsIdentityPath(...segments: string[]): string {
  return segments
    .map((segment) => normalizeSegment(String(segment ?? "")))
    .filter((segment) => segment.length > 0)
    .join("/");
}

export function buildUnsRoutePath(...segments: string[]): string {
  return `/${buildUnsIdentityPath(...segments)}`.replace(/\/{2,}/g, "/");
}

File: packages/uns-py/src/uns_kit/uns_path.py
from __future__ import annotations


def _normalize_segment(value: str) -> str:
    return str(value or "").strip().strip("/")


def build_uns_identity_path(*segments: str) -> str:
    return "/".join(
        segment
        for segment in (_normalize_segment(str(value)) for value in segments)
        if segment
    )


def build_uns_route_path(*segments: str) -> str:
    return f"/{build_uns_identity_path(*segments)}".replace("//", "/")
